duckduckgo/iOS
# Autofill fix to prevent saving new login when device passcode disabled

…abled and no logins saved



Task/Issue URL: https://app.asana.com/0/414235014887631/1203534198782091/f
Tech Design URL:
CC: @THISISDINOSAUR 

**Description**:
Fix for edge case introduced by https://github.com/duckduckgo/iOS/pull/1395 so that when a user does not have any device authentication enabled they cannot create any new logins and are presented with the correct "Set up your device passcode to use Autofill" screen



**Steps to test this PR**:
1. From a clean install, disable device passcode and try to access the Autofill logins screen. You should be presented with the "Set up your device passcode to use Autofill" screen

For regression testing:
1. Enable device passcode and try again to access the Autofill logins screen. This time you should be able to access the empty version of the screen without auth challenge as per https://app.asana.com/0/1201462886803403/1203328956136997/f and should be able to create a new login
2. Exit out of the Autofill screens, then come back in. This time you should be prompted to authenticate before being able to access the autofill logins 
3. Exit out of the Autofill screens again and disable device passcode. Try to access Autofill logins again and you should be presented with the "Set up your device passcode to use Autofill" screen



**Copy Testing**:


**Orientation Testing**:


**Device Testing**:


**OS Testing**:


**Theme Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
